modules/trainer/GenericTrainer.py

```diff
@@ -632,6 +641,68 @@ def sample_commands_fun():
 
                     self.one_step_trained = True
 
+                    # validation
+                    self.data_loader_validation.get_data_set().start_next_epoch()
```

Comment:
`self.data_loader_validation` might not exist. This whole block of code needs a check before it can be executed.

actually, I think it also makes sense to move this whole validation logic into a new function that's just called here if `config.validation` is true